### ISSUE TITLE
Import from a virtual disk in an external SSD is slow at roughly 5-9MBps inside a VM.

### ISSUE THREAD

**Issue by cpauya** [learningequality/kolibri#6715]
### Observed behavior
Inside an Ubuntu 16.04.6 VM - Importing channel content (Khan Academy English about 19GB) from a local virtual disk is slow at roughly 5-9MBps whereas copying a folder of the same size yields >100MBps inside the same VM.  

This local disk is a "virtual disk" that is saved in an [external Seagate Fast SSD 500GB](https://www.techradar.com/reviews/seagate-fast-ssd-500gb) that's supposed to have 540MBps read/write theoretical speeds.  It's connected via my USB Type-C to get the best performance.

Progress so far:
* 35 minutes = 30% 
* 50 minutes = 37%
* 55 minutes = 40%

At this rate, it'll take about 135+ minutes to finish importing the 19GB of channel content.  

Is this normal behaviour or is there a bottleneck somewhere when importing in Kolibri? 

Here's the .gif of the Ubuntu VM with `iotop` while importing
![kolibri--import content from external ssd 3](https://user-images.githubusercontent.com/175580/77935915-b3085880-72e4-11ea-9961-1a62fc975119.gif)

Here's the .gif of the Ubuntu VM with `iotop` while copying the folder inside the same VM
![kolibri--import content from external ssd 5](https://user-images.githubusercontent.com/175580/77936083-f5ca3080-72e4-11ea-9ef4-cb7f5c5479d5.gif)

### Expected behavior
I expected a faster than 9MBps read/write speed when importing content from the local virtual disk (saved in an external SSD) even when inside an Ubuntu VM.

### Context
* Kolibri version: [v0.13.2-beta1](https://github.com/learningequality/kolibri/releases/tag/v0.13.2-beta1)
* Operating system: Ubuntu 16.04.6 inside VMWare Fusion on a macOS Mojave 10.14.6
* Virtual RAM: 8GB
* Virtual Processors: 2
* Shared Graphics Memory: 768MB (as recommended by VMWare)
* Browser: Firefox v74.0

### Related Issue/s:

This seems related to this issue [Khan Academy (English) Import Metrics](https://github.com/learningequality/kolibri/issues/6684).

### Note/s

* I just used the results from `iotop` for the 5-9MBps speed.  I have no other means (yet) to determine why the import process is slower than I expected. 😸

**Comment by rtibbles** [comment]
Similar issues arise for downloads - downloading content also fails to get anywhere near to maxing out available bandwidth.

Could you compare running the same process from the command line using the management command? My intuition is that it is to do with context switching because we are using threads exclusively for our task management atm.

**Comment by cpauya** [comment]
I tried it by running `kolibri start` in the Terminal. 

Progress:
* 6 minutes = 12%, 18% CPU; write speeds up to 16MBps
* 10 minutes = 15%; 28% CPU; 232MB RAM; write speeds up to 16MBps
* 22 minutes = 25%; 38% CPU; 239MB RAM; write speeds up to 17MBps
* 32 minutes = 30%; 40% CPU; 241MB RAM; write speeds up to 7MBps
* 35 minutes = 32%; 39% CPU; 241MB RAM; write speeds up to 10MBps
* 50 minutes = 40%; 41% CPU; 247MB RAM; write speeds up to 10MBps
* 61 minutes = 44%; 42% CPU; 248MB RAM; write speeds up to 6MBps
* 74 minutes = 50%; 42% CPU; 250MB RAM; write speeds up to 5MBps 
* 138 minutes = 78%; 44% CPU; 257MB RAM; write speeds up to 7MBps == 176 minutes DONE

At this rate the whole 19GB import will take 148+ minutes.  It starts fast then eventually degrades. 

Note: I think Kolibri in Ubuntu is already using `kolibri start`?  I tried `kolibri manage runserver` but it's stuck:

<img width="1552" alt="Screenshot 2020-03-31 14 52 30" src="https://user-images.githubusercontent.com/175580/77995973-58601280-735f-11ea-8529-4e513ba43092.png">

I will file a separate issue on the "stuck" `runserver` management command (if that's indeed an issue).

**Comment by rtibbles** [comment]
Oh, I meant invoking the import by doing
```
kolibri manage importchannel disk <channel_id> </path/to/disk/>
kolibri manage importcontent disk <channel_id> </path/to/disk/>
```

**Comment by cpauya** [comment]
Yikes, I misunderstood!  I will try that next. 😸

**Comment by cpauya** [comment]
Followed guide on how to [Import Content Channels from Internet](https://kolibri.readthedocs.io/en/latest/manage/command_line.html#import-content-channels-from-internet) but encountered this on my Ubuntu VM:

This is for Khan Academy (English), tried it multiple times but only got the warning of version already exists:
```bash
user@ubuntu:~$ kolibri manage importchannel network 1ceff53605e55bef987d88e0908658c5
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Invoking command importchannel network 1ceff53605e55bef987d88e0908658c5
INFO     Downloading data for channel id 1ceff53605e55bef987d88e0908658c5
  [######################################################################]  100%WARNING  Version 14 of channel 1ceff53605e55bef987d88e0908658c5 already exists in database; cancelling import of version 14
INFO     Channel 1ceff53605e55bef987d88e0908658c5 successfully imported into the database
```

I tried "How to get started with Kolibri" twice (results below) but the same results as above but of different version.
```bash
user@ubuntu:~$ kolibri manage importchannel network 624e09bb5eeb4d20aa8de62e7b4778a0
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Invoking command importchannel network 624e09bb5eeb4d20aa8de62e7b4778a0
INFO     Downloading data for channel id 624e09bb5eeb4d20aa8de62e7b4778a0
  [######################################################################]  100%WARNING  Version 23 of channel 624e09bb5eeb4d20aa8de62e7b4778a0 already exists in database; cancelling import of version 23
INFO     Channel 624e09bb5eeb4d20aa8de62e7b4778a0 successfully imported into the database
```

What could be wrong?  Do I need to reset my database?

**Comment by rtibbles** [comment]
In this case, you can just skip to the second step, if the database is already there.

**Comment by cpauya** [comment]
Now I have another error with the `importcontent` command:

```bash
user@ubuntu:~$ kolibri manage importcontent network 1ceff53605e55bef987d88e0908658c5
INFO     Running Kolibri with the following settings: kolibri.deployment.default.settings.base
INFO     Invoking command importcontent network 1ceff53605e55bef987d88e0908658c5
Traceback (most recent call last):
  File "/usr/bin/kolibri", line 11, in <module>
    load_entry_point('kolibri==0.13.2b1', 'console_scripts', 'kolibri')()
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 243, in invoke
    return super(KolibriDjangoCommand, self).invoke(ctx)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3/dist-packages/kolibri/utils/cli.py", line 641, in manage
    execute_from_command_line(["kolibri manage"] + ctx.args)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python3/dist-packages/kolibri/core/tasks/management/commands/base.py", line 102, in handle
    return self.handle_async(*args, **options)
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 439, in handle_async
    renderable_only=options["renderable_only"],
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 159, in download_content
    renderable_only=renderable_only,
  File "/usr/lib/python3/dist-packages/kolibri/core/content/management/commands/importcontent.py", line 257, in _transfer
    total=total_bytes_to_transfer + dummy_bytes_for_annotation
  File "/usr/lib/python3/dist-packages/kolibri/core/tasks/management/commands/base.py", line 115, in start_progress
    total=total, level=level, update_callback=self._update_all_progress
  File "/usr/lib/python3/dist-packages/kolibri/core/tasks/management/commands/base.py", line 38, in __init__
    self.progressbar = click.progressbar(length=total, width=0)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/termui.py", line 351, in progressbar
    width=width, color=color)
  File "/usr/lib/python3/dist-packages/kolibri/dist/click/_termui_impl.py", line 81, in __init__
    iterable = range_type(length)
TypeError: 'float' object cannot be interpreted as an integer
user@ubuntu:~$
```

This seems related to this already closed issue - [error during command line import on large channels](https://github.com/learningequality/kolibri/issues/6597).

**Comment by rtibbles** [comment]
It's related, but a little different - seems like we should be coercing the `total` value to an int as well. Could you open a separate issue for this error, as I don't think we should release with it.

**Comment by indirectlylit** [comment]
see also: https://github.com/learningequality/kolibri/issues/6684

**Comment by rtibbles** [comment]
This should be fixed by #7521.
